qt/python/instrumentview/instrumentview/test/test_presenter.py

```
@@ -586,25 +586,51 @@ def test_sample_position_mesh_added(self, mock_create_sample_mesh):
         self._mock_view.add_rgba_mesh.assert_not_called()
 
+    def test_create_and_add_sample_mesh_with_shape_creates_polydata(self):
+        self._mock_view.is_show_sample_position_checkbox_checked.return_value = True
+        self._model._sample_position = np.array([0.0, 0.0, 0.0])
+        # Two triangles: shape (2, 3, 3) — 2 triangles, each with 3 vertices of xyz
+        self._model._sample_shape = np.array([[[0, 0, 0], [1, 0, 0], [0, 1, 0]], [[0, 0, 0], [0, 1, 0], [0, 0, 1]]], dtype=float)
+        self._presenter._create_and_add_sample_mesh()
+        self._mock_view.add_rgba_mesh.assert_called_once()
+        mesh_arg = self._mock_view.add_rgba_mesh.call_args.args[0]
+        self.assertEqual(self._mock_view.add_rgba_mesh.call_args.kwargs["scalars"], "colours")
+        # 2 triangles × 3 vertices = 6 vertices
```

Ruff (0.15.18)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)